test/e2e/framework/pod/wait.go

```diff
@@ -362,6 +363,80 @@ func WaitForAllPodsCondition(c clientset.Interface, ns string, opts metav1.ListO
 	return pods, maybeTimeoutError(err, "waiting for at least %d pods to be %s (matched %d)", minPods, conditionDesc, matched)
 }
 
+// WaitForPodsRunning waits for a given `timeout` to evaluate if a certain amount of pods in given `ns` are running.
+func WaitForPodsRunning(c clientset.Interface, ns string, num int, timeout time.Duration) error {
+	matched := 0
+	err := wait.PollImmediate(poll, timeout, func() (done bool, err error) {
+		pods, err := c.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return handleWaitingAPIError(err, true, "listing pods")
+		}
+		var nonMatchingPods []string
+		for _, pod := range pods.Items {
+			ready, _ := testutils.PodRunningReady(&pod)
+			if !ready {
+				nonMatchingPods = append(nonMatchingPods, identifier(&pod))
+			}
+		}
+		if len(nonMatchingPods) <= 0 {
```

Comment:
the functions says `evaluate if a certain amount of pods in given `ns` are running.`, but this doesn't checked yet that there are `num`pods running.
Maybe you can exit fast `if len(pods) < num`

Reply:
oh, thanks for the catch. I should have rewritten it to the style in WaitForPodsSchedulingGated() but somehow missed it. Let me update.

Reply:
updated.